examples/Extensions/RemoveEntireSitelinkCampaignExtensionSetting.php

```diff
@@ -0,0 +1,233 @@
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the client customer ID
+     * @param int $campaignId the campaign ID
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $campaignId
+    ) {
+        $extensionFeedItemResourceNames =
+            self::getAllSitelinkExtensionFeedItems($googleAdsClient, $customerId, $campaignId);
+
+        // 1) Removes the campaign extension setting.
+
+        // Creates the resource name of a campaign extension setting to remove.
+        $campaignExtensionSettingResourceName = ResourceNames::forCampaignExtensionSetting(
+            $customerId,
+            $campaignId,
+            ExtensionType::SITELINK
+        );
+
+        // Creates a campaign extension setting operation.
+        $extensionFeedItemOperation = new CampaignExtensionSettingOperation();
+        $extensionFeedItemOperation->setRemove($campaignExtensionSettingResourceName);
+
+        // Issues a mutate request to remove the campaign extension setting.
+        $extensionFeedItemServiceClient =
+            $googleAdsClient->getCampaignExtensionSettingServiceClient();
+        $response = $extensionFeedItemServiceClient->mutateCampaignExtensionSettings(
+            $customerId,
+            [$extensionFeedItemOperation]
+        );
+
+        // Prints the resource name of the removed campaign extension setting.
+        /** @var CampaignExtensionSetting $removedCampaignExtensionSetting */
+        $removedCampaignExtensionSetting = $response->getResults()[0];
+        printf(
+            "Removed a campaign extension setting with resource name: '%s'.%s",
+            $removedCampaignExtensionSetting->getResourceName(),
+            PHP_EOL
+        );
+
+        // 2) Removes all the extension feed items of the previously removed campaign extension
+        // setting.
+
+        foreach ($extensionFeedItemResourceNames as $extensionFeedItemResourceName) {
+            // Creates an operation to remove the extension feed item.
```

Comment:
Is it mandatory to execute N requests of 1 operation here? 1 request of N operations seems better.

Reply:
Obsolete due to the new way of implementation.